appintro/src/main/java/com/github/paolorotolo/appintro/AppIntro2.kt

```diff
@@ -3,23 +3,23 @@ package com.github.paolorotolo.appintro
 import android.graphics.drawable.Drawable
 import android.os.Bundle
 import android.view.View
-import android.widget.FrameLayout
 import android.widget.ImageButton
-import android.widget.LinearLayout
 import androidx.annotation.ColorInt
+import android.widget.FrameLayout
+import androidx.constraintlayout.widget.ConstraintLayout
 
 abstract class AppIntro2 : AppIntroBase() {
 
     var backgroundView: View? = null
         set(value) {
             field = value
             if (field != null) {
-                backgroundFrame.addView(field)
+                backgroundFrame.addView(field, 0)
```

Comment:
So this is probably not working (or at least will have some side effects).
Like in `CustomBackgroundIntro.java`, we call that passing an `ImageView` with layout params set like:

```
imageView.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
```

So this will luckily work in our case. But if you customize `FrameLayout.LayoutParams`, it might not work properly as now is consumer by a `ConstraintLayout` and will work properly only with ` ConstraintLayout.LayoutParams`.

So the alternatives here are:

1) Change the variable to:
`var backgroundDrawable: Drawable? = null`
And call `backgroundFrame.background = ...` on the setter

2) Make sure we set the proper LayoutParam we expect on the view. In this case we will override what the developers will set. Like:
```
if (field != null) {
                field?.layoutParams = ConstraintLayout.LayoutParams()
                backgroundFrame.addView(field, 0)
            }
```

I'd advice for the first one. Is a breaking change but is making everything cleaner.

Reply:
Okay. I also think 1st one would be better. Alright then Sold.^_^

Reply:
Done